src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplay/ContentItemDisplayCoordinator.cs

```diff
@@ -90,6 +92,35 @@ public async Task BuildDisplayAsync(ContentItem contentItem, BuildDisplayContext
                         }
                     }
 
+                    var tempContext = context;
```

Comment:
Nice, I like this `tempContext` trick that you've used to get fields of dynamic, explicit parts to belong to part shape